homeassistant/components/automation/event.py

```diff
@@ -29,12 +29,12 @@ def async_trigger(hass, config, action):
     event_type = config.get(CONF_EVENT_TYPE)
     event_data = config.get(CONF_EVENT_DATA)
 
-    @asyncio.coroutine
+    @async_safe
```

Comment:
Also, for typing, callback functions are Callable's, per https://docs.python.org/3.5/library/typing.html#callable (personally I think they should have a separate Callback type, to avoid confusion with any old callable...).